src/SceneKit/SCNCompat.cs

```diff
@@ -72,4 +76,28 @@ public unsafe virtual Task PresentSceneAsync (SCNScene scene, global::XamCore.Sp
 	}
 #endif
 #endif
+
+#if WATCH || XAMCORE_4_0
+	[Watch (4,0)]
+	[Mac (10,9), iOS (8,0)]
+	public delegate void SCNAnimationEventHandler (ISCNAnimationProtocol animation, NSObject animatedObject, bool playingBackward);
+#else
+	[Mac (10,9), iOS (8,0)]
+	public delegate void SCNAnimationEventHandler (CAAnimation animation, NSObject animatedObject, bool playingBackward);
+#endif
+
+	public partial class SCNAnimationEvent : NSObject {
+		public static SCNAnimationEvent Create (nfloat keyTime, SCNAnimationEventHandler eventHandler)
+		{
+			var handler = new Action<IntPtr, NSObject, bool> ((animationPtr, animatedObject, playingBackward) => {
+#if WATCH || XAMCORE_4_0
+				var animation = Runtime.GetINativeObject<ISCNAnimationProtocol> (animationPtr, true);
+#else
+				var animation = Runtime.GetINativeObject<CAAnimation> (animationPtr, true);
```

Comment:
might be smaller to do something like
```
#if WATCH
using AnimationType = ISCNAnimationProtocol;
#else
using AnimationType = CAAnimation;
#endif
```
at the top and drop the other conditionals.